### PR TITLE
feat(substrate-bindings): `Binary.fromOpaqueBytes`

### PR DESCRIPTION
**Pull request by josepot**
It's very annoying when using runtime-apis that expect an extrinsic as an input to have to manually remove its length. I think that it makes sense to add these 2 static functions.

### CODE REVIEW

**Review by carlosala — APPROVED** [review]
I definitely agree, this is the way to proceed 👍👍
Thanks! 😉🚀

**Comment by josepot** [comment]
I made a few more changes, and I also added an E2E test. Please, review it again. 🙏

File: packages/substrate-bindings/src/codecs/scale/Binary.ts [hunk 1]
```diff
@@ -1,38 +1,69 @@
...
+      } catch (_) {}
+      throw new Error("Invalid opaque bytes")
+    } else this.#bytes = data
   }
 
-  asText = () =>
-    this.#str === null
-      ? (this.#str = textDecoder.decode(this.#bytes))
-      : this.#str
+  asText = () => (this.#str ||= textDecoder.decode(this.#bytes))
```
Comment (carlosala) [inline packages/substrate-bindings/src/codecs/scale/Binary.ts:38]: I'd say it's safer to use `??=` instead. If the array is empty it defeats its purpose, for example.
Idem for the rest of the file.

**Review by carlosala — APPROVED** [review]
I like it! Another option would be to use a flag in the methods, something like `foo.asBytes({ opaque: true })`.
Both options seem nice to me, it can be merged as is as well!

File: packages/substrate-bindings/src/codecs/scale/Binary.ts (hunk 1 above)
Comment (josepot) [reply packages/substrate-bindings/src/codecs/scale/Binary.ts:38]: yep, good point!

**Comment by carlosala** [comment]
You can use `??=` in the rest of the file as well 👍👍

**Review by voliva — APPROVED** [review]

**Pull request merged**